source/base/parameter_handler.cc

```diff
@@ -516,19 +633,45 @@ namespace Patterns
 
 
 
-  std::string List::description () const
+  std::string List::description (const OutputStyle style) const
   {
-    std::ostringstream description;
+    switch (style)
+      {
+      case Machine:
+      {
+        std::ostringstream description;
 
-    description << description_init
-                << " list of <" << pattern->description() << ">"
-                << " of length " << min_elements << "..." << max_elements
-                << " (inclusive)";
-    if (separator != ",")
-      description << " separated by <" << separator << ">";
-    description << "]";
+        description << description_init
+                    << " list of <" << pattern->description(style) << ">"
+                    << " of length " << min_elements << "..." << max_elements
+                    << " (inclusive)";
+        if (separator != ",")
+          description << " separated by <" << separator << ">";
+        description << "]";
 
-    return description.str();
+        return description.str();
+      }
+      case Text:
+      case LaTeX:
+      {
+        std::ostringstream description;
+
+        description << "a list of "
+                    << min_elements << " to " << max_elements
+                    << " elements ";
+        if (separator != ",")
+          description << "separated by <" << separator << "> ";
+        description  << "where each element is "
+                     << pattern->description(style);
```

Comment:
Since lists can be nested, I think it would be useful to have some kind of parentheses or braces or brackets or something else around the `pattern->description(style)`. Otherwise, you could get things such as
```
  A list of 3 to 5 elements separated by <,> where each element is a list of 3 to 5 elements separated
  by <:> where each element is $SOMETHING_COMPLICATED.
```
That's not grammatically wrong, but may be difficult to read.